sources/vega/snapshot/v1/snapshot.proto

```diff
@@ -553,9 +553,25 @@ message StakeVerifierPending {
   string id = 8;
 }
 
-message Topology{
+message KeyRotation {
+  uint64 block_height = 1;
+  string node_id = 2;
+  string new_pub_key = 3;
+  string old_pub_key = 4;
+}
+
+message PendingKeyRotation {
+  uint64 block_height = 1;
+  string node_id = 2;
+  string new_pub_key = 3;
+  uint32 new_key_number = 4;
```

Comment:
I think we really need to be consistent here. Both BIP-32 and slip10 refer to this attribute of a child key as an index, so we should call this field `new_key_index`. Same in `message ValidatorUpdate` it should be  `vega_pub_key_index`.